### ISSUE TITLE
—help is broken when default is equal to None on a option.

### ISSUE THREAD

**Issue by caner-cetin** [bckohan/django-typer#41]
I am on mobile so sorry for limited example

`foo: Annotated[Optional[str], typer.Option(default=None)`

register any command with an option like this and try to run —help for that function. it always errors out, due to click trying to parse param declarations and param declarations being none in this case. and default cannot be None when param declaration is not None due to overloads. help?

**Comment by bckohan** [comment]
Hi @caner-cetin , thanks for the report.

Do this instead:

```python
foo: Annotated[Optional[str], typer.Option()] = None
```
from typer 0.8 to 0.9 passing default to Option does not seem to work anymore.

Let me know if that solved your issue!

**Comment by caner-cetin** [comment]
That also quite doesnt work.  Password is optional in this case
```python
        password: Annotated[
            Optional[str],
            typer.Option(
                "--password", hide_input=True,prompt=True
            ),
        ] = None,
``` and I keep spamming enter with blank input:
<img width="445" alt="resim" src="https://github.com/bckohan/django-typer/assets/92731060/3664e56b-475e-49a3-91eb-c39cebf96459">
Yet I cant go further. Pressing ESC then enter yields `\x1b` character, of course. What can I do?

**Comment by bckohan** [comment]
So just hitting enter will reprompt - that particular behavior is expected. But there is a bug here. If you enter the password twice it will work - the bug is that it prompts you twice #42 

Thanks for bringing this to my attention!

**Comment by caner-cetin** [comment]
And for now, what do you recommend for truly Optional options? Because clearly our examples doesnt work right now. I can only give `" "` or ESC's unicode representation right now to option for skipping. Blank input is out of the window until a fix.

**Comment by bckohan** [comment]
They work if prompt is not True.

**Comment by bckohan** [comment]
The double prompt bug is fixed in 1.0.5

**Comment by caner-cetin** [comment]
thank you <3
